test/RibbonThetaVault.js

```diff
@@ -2426,6 +2426,98 @@ function behavesLikeRibbonOptionsVault(params) {
       });
     });
 
+    describe("#withdrawLater", () => {
+      time.revertToSnapshotAfterEach();
+
+      it("rejects a withdrawLater of 0 shares", async function () {
+        await expect(
+          this.vault.withdrawLater(BigNumber.from("0"))
+        ).to.be.revertedWith("!shares");
+      });
+
+      it("rejects a completeScheduledWithdrawal if nothing scheduled", async function () {
+        const depositAmount = BigNumber.from("100000000000");
+        await depositIntoVault(
+          params.collateralAsset,
+          this.vault,
+          depositAmount
+        );
+
+        await expect(
+          this.vault.completeScheduledWithdrawal()
+        ).to.be.revertedWith("Scheduled withdrawal not found");
+      });
+
+      it("rejects a scheduled withdrawal when greater than balance", async function () {
+        const depositAmount = BigNumber.from("100000000000");
+        await depositIntoVault(
+          params.collateralAsset,
+          this.vault,
+          depositAmount
+        );
+
+        //way too much
+        await expect(
+          this.vault.withdrawLater(BigNumber.from("200000000000"))
+        ).to.be.revertedWith("Insufficient shares");
+
+        // barely too much
+        await expect(
+          this.vault.withdrawLater(BigNumber.from("100000000001"))
+        ).to.be.revertedWith("Insufficient shares");
+      });
+
+      it("accepts a withdrawLater if less than or equal to balance", async function () {
+        var balanceBeforeWithdraw;
+        const depositAmount = BigNumber.from("200000000000");
+        await depositIntoVault(
+          params.collateralAsset,
+          this.vault,
+          depositAmount
+        );
+
+        await this.vault.withdrawLater(BigNumber.from("100000000000"));
+
+        await this.rollToNextOption();
+
+        if (params.collateralAsset === WETH_ADDRESS) {
+          balanceBeforeWithdraw = await provider.getBalance(user);
+        } else {
+          balanceBeforeWithdraw = await this.assetContract.balanceOf(user);
+        }
+        const vaultBalanceBeforeWithdraw = await this.assetContract.balanceOf(
+          this.vault.address
+        );
+
+        // Queued withdrawals + 10% of available assets set aside
+        assert.equal(
+          vaultBalanceBeforeWithdraw.toString(),
+          BigNumber.from("110000000000").toString()
+        );
+
+        const tx = await this.vault.completeScheduledWithdrawal({
+          gasPrice,
+        });
```

Comment:
Let's add a few more checks here, we need to:

1. Check that the `ScheduledWithdrawCompleted` event is emitted correctly
2. Check that `scheduledWithdrawals[msg.sender]` is set back to 0
3. Check that the vault's balance is correctly deducted